src/tests/ftest/io/ior_intercept_verify_data_integrity.py

```diff
@@ -0,0 +1,161 @@
+        self.lock = threading.Lock()
+
+    def test_ior_intercept_verify_data(self):
+        """Jira ID: DAOS-3502.
+
+        Test Description:
+            Purpose of this test is to run ior through dfuse with
+            interception library  on 5 clients and without interception
+            library on 1 client for at least 30 minutes and verify the
+            data integrity using ior's Read Verify and Write Verify
+            options.
+
+        Use case:
+            Run ior with read, write, fpp, read verify
+            write verify for 30 minutes
+            Run ior with read, write, read verify
+            write verify for 30 minutes
+
+        :avocado: tags=all,daosio,hw,full_regression,iorinterceptverifydata
+        """
+        intercept = os.path.join(self.prefix, 'lib64', 'libioil.so')
+        with_intercept = dict()
+        self.run_multiple_ior_with_pool(with_intercept, intercept)
+
+        IorCommand.log_metrics(self.log, "5 clients - with " +
+                               "interception library", with_intercept[1])
+        IorCommand.log_metrics(self.log, "1 client - without " +
+                               "interception library", with_intercept[2])
+
+    def run_multiple_ior_with_pool(self, results, intercept=None):
+        """Execute ior with optional overrides for ior flags and object_class.
+
+        If specified the ior flags and ior daos object class parameters will
+        override the values read from the yaml file.
+
+        Args:
+            intercept (str): path to the interception library. Shall be used
+                             only for POSIX through DFUSE.
+            ior_flags (str, optional): ior flags. Defaults to None.
+            object_class (str, optional): daos object class. Defaults to None.
+        """
+        # Create a pool if one does not already exist
+        if self.pool is None:
+            self.create_pool()
+        # Update IOR params with the pool
+        self.ior_cmd.set_daos_params(self.server_group, self.pool)
+
+        # start dfuse if api is POSIX
+        if self.ior_cmd.api.value == "POSIX":
+            # Connect to the pool, create container and then start dfuse
+            # Uncomment below two lines once DAOS-3355 is resolved
+            # self.pool.connect()
+            # self.create_cont()
+            self.start_dfuse()
+
+        # Create two jobs and run in parallel.
+        # Job1 will have 3 client set up to use dfuse + interception
+        # library
+        # Job2 will have 1 client set up to use only dfuse.
+        job1 = self.get_new_job(self.hostlist_clients[:-1], 1,
+                                results, intercept)
+        job2 = self.get_new_job([self.hostlist_clients[-1]], 2,
+                                results, None)
+
+        job1.start()
+        job2.start()
+        job1.join()
+        job2.join()
+
+    def get_new_job(self, clients, job_num, results, intercept=None):
```

Comment:
As this function is also being used in PR#1619, I would suggest to centralize this function somewhere. May be ior_test_base.py or even a separate util file for interception whatever you feel suited.

Reply:
Agreed. But since both PR#1619 and this one are done in parallel, there's a dependency. One needs to go into the master before I can push another one for review which causes delay. I'd rather create a story to make this change in the future.